course_catalog/migrations/0083_course_feature_tags.py

```diff
@@ -2,19 +2,29 @@
 
 import django.contrib.postgres.fields.jsonb
 from django.db import migrations
+from django.core.paginator import Paginator
 
 from course_catalog.constants import PlatformType
 
 
+def chunked_iterator(queryset):
+    paginator = Paginator(queryset, 100)
+    for page in range(1, paginator.num_pages + 1):
+        for obj in paginator.page(page).object_list:
+            yield obj
+
+
 def set_feature_tags(apps, schema_editor):
     """
     Save feature tags for every OCW course based on the run's raw_json
     """
     Course = apps.get_model("course_catalog", "Course")
     LearningResourceRun = apps.get_model("course_catalog", "LearningResourceRun")
-    for run in LearningResourceRun.objects.filter(
-        platform=PlatformType.ocw.value
-    ).iterator():
+    for run in chunked_iterator(
+        LearningResourceRun.objects.filter(platform=PlatformType.ocw.value)
+        .all()
+        .order_by("id")
+    ):
         course = Course.objects.get(id=run.object_id)
         if run:
             course.course_feature_tags = [
```

Comment:
I'm seeing an exception near this line. Can you add some code to handle the case where `raw_json` is `None`?

      File "./manage.py", line 14, in <module>
        execute_from_command_line(sys.argv)
      File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
        utility.execute()
      File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
        self.fetch_command(subcommand).run_from_argv(self.argv)
      File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
        self.execute(*args, **cmd_options)
      File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
        output = self.handle(*args, **options)
      File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
        res = handle_func(*args, **kwargs)
      File "/usr/local/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 234, in handle
        fake_initial=fake_initial,
      File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 117, in migrate
        state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
      File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
        state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
      File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
        state = migration.apply(state, schema_editor)
      File "/usr/local/lib/python3.7/site-packages/django/db/migrations/migration.py", line 124, in apply
        operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
      File "/usr/local/lib/python3.7/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
        self.code(from_state.apps, schema_editor)
      File "/src/course_catalog/migrations/0083_course_feature_tags.py", line 32, in set_feature_tags
        for tag in run.raw_json.get("course_feature_tags", [])
    AttributeError: 'NoneType' object has no attribute 'get'

Reply:
Sure, though it should never be None on CI/RC/prod.